client/src/Components/Townbox/index.tsx

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable react/prop-types */
+import React from 'react'
+import './index.css'
+import {Button} from '../Button'
+
+export interface TownboxProps {
+  /**
+   * Header colour: enter either orange/primary
+   */
+  headerColor?: 'primary' | 'orange'
+
+  /**
+   * Heading contents
+   */
+  headerText?: string
+
+  /**
+   * Textbox contents
+   */
+  captionText?: string
+
+  /**
+   * Show visited button?
+   */
+  showButton?: boolean
+
+  /**
+   * specify width in terms of px
+   */
+  widthSize?: number
+}
+
+/**
+ * Primary UI component for user interaction
+ */
+export const Townbox: React.FC<TownboxProps> = ({
+  headerText = 'headerText',
+  captionText = 'captionText',
+  headerColor = 'orange',
+  showButton = false,
+  widthSize = 550
+}) => {
+  return (
+    <div className = "flex" style={{width: widthSize}}>
+      
+      <h2 className = {`font-serif text-white p-2 px-4 border-black border-4 rounded-md ml-10 mt-4 absolute z-10 bg-${headerColor}`}
+      > {headerText} </h2>
+      
+      <div className = "captionboxborder h-64 absolute mt-12" style={{width: widthSize}}></div> 
+        
+      <div className = "captionbox place-self-center p-2 mt-12 border-black flex relative flex-col items-center justify-center w-full h-64">
+        <p className = "font-sans border-solid pb-2 mb-0 z-10 text-2xl relative mt-4 break-words leading-8" style={{width: widthSize - 100}}>
+          {captionText}
+        </p>
+            
+        {/* visit button is optional*/}
+        {showButton && <Button label="Visit" size="small" backgroundColor='#00acd0' primary={false}/>} 
```

Comment:
This should either be orange or purple, you can use the header colour to decide which should be displayed

Reply:
I think after this it should be good to merge in

Reply:
@Drewbi Hi Drew just one question, should I be assuming that the Button component I'm using is Niall's button component or the button component that is implemented in the main branch? Because the props that I'm passing into the Button will be different between the two.

Reply:
Just do whatever will work with the current main branch and don't worry about trying to anticipate changes.

Reply:
Alright no problem. The change is now implemented in the branch. However, one concern is that the button component does not allow the text colour to be changed, so when the background of the button is changed to purple, it's pretty hard to read the text. Not sure how to overcome this without making changes to the button component itself.

Reply:
That's no problem as the buttons will have the same white text as the header so they should be just as readable :)